### PR TITLE
fix(hostinfo/sysctl): Fix broken sysctl check, and refactor to work o…

### PR DESCRIPTION
**Pull request by kaustavha**
…n execfiletobuffer

### CODE REVIEW

File: hostinfo/sysctl.lua [hunk 1]
```diff
@@ -29,39 +29,33 @@ end
...
-  child:on('error', function(err)
-    self._error = err
-    callback()
+  execFileToBuffers('which', {'sysctl'}, {}, function(err, exitcode, data, stderr)
+    if exitcode ~= 0 then
+      self._error = fmt("sysctl exited with a %d exit_code", exitcode)
+      return callback()
+    end
+    -- Remove /n at the end of data returned from which and call
+    return execFileToBuffers(data:sub(1, data:len()-1), {'-A'}, {}, execCb)
```
Comment (rphillips) [inline hostinfo/sysctl.lua]: there is a misc.trim function

Comment (kaustavha) [inline hostinfo/sysctl.lua]: screen shot 2015-08-04 at 3 33 33 pm
virgo/util/misc .trim doesnt trim the way i want, over here im trying to remove a /n at the end of the line. 
perhaps i can extend the trim function instead

**Comment by kaustavha** [comment]
Relies on:
 #827 
virgo-agent-toolkit/virgo-base-agent#172 
virgo-agent-toolkit/virgo-base-agent#173

**Comment by rphillips** [comment]
```lua
local execFileToBuffers = require('./misc').execFileToBuffers
```

**Pull request closed without merging**